### PR TITLE
Feature sizing loop backtracking

### PR DESCRIPTION
**Pull request by jmvegh**
added backtracking and bounding features to the sizing loop framework

### CODE REVIEW

File: regression/scripts/AVL/avl_files/aircraft.avl [hunk 1]
```diff
@@ -62,7 +62,7 @@ SECTION
 
 SECTION
 #Xle    Yle      Zle      Chord     Ainc  Nspanwise  Sspace
-32.4164746575  1.54    7.777    0.95    0.0     
+32.4164746575  0.0    7.777    0.95    0.0
```
Comment (WalterMaier) [inline regression/scripts/AVL/avl_files/aircraft.avl]: ??? @mclarke2 needs to look at this....

File: regression/scripts/sizing_loop/sizing_loop.py [hunk 2]
```diff
@@ -47,24 +50,44 @@ def main():
...
+    check_read_res = -0.06783837567842196
+    error_res      = (data_outputs[1]-check_read_res)/check_read_res
+    
+    #remove files for later
+    os.remove('sizing_outputs.txt')
+    os.remove('y_err_values.txt')
     print 'error = ', error
-    assert(error<1e-6), 'sizing loop regression failed'    
+    print 'error_res = ', error_res
+    assert(error<1e-5), 'sizing loop regression failed'
```
Comment (WalterMaier) [inline regression/scripts/sizing_loop/sizing_loop.py]: reason for 1e-5?

File: trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py [hunk 3]
```diff
@@ -308,6 +308,20 @@ def build_surrogate(self):
...
+                CD_sur[ii,jj] = cd_surrogate.predict([np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]])])
+        
+        fig = plt.figure('Coefficient of Lift Surrogate Plot')    
+        plt_handle = plt.contourf(AoA_mesh/Units.deg,mach_mesh,CL_sur,levels=None)
+        cbar = plt.colorbar()
+        plt.scatter(xy[:,0]/Units.deg,xy[:,1])
+        plt.xlabel('Angle of Attack (deg)')
+        plt.ylabel('Mach Number')
+        cbar.ax.set_ylabel('Coefficient of Lift')
+
```
Comment (WalterMaier) [inline trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py]: Should this be added?  Would these plots be regressed?

File: trunk/SUAVE/Analyses/Stability/AVL.py [hunk 4]
```diff
@@ -442,11 +442,21 @@ def build_surrogate(self):
...
-                NP_sur[ii,jj]    = neutral_point_surrogate.predict(np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]]))                
+                CM_sur[ii,jj]    = cm_surrogate.predict([np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]])])
+                Cm_a_sur[ii,jj]  = cm_alpha_surrogate.predict([np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]])])
+                Cn_b_sur[ii,jj]  = cn_beta_surrogate.predict([np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]])])
+                NP_sur[ii,jj]    = neutral_point_surrogate.predict([np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]])])                
                 
+
+        fig = plt.figure('Coefficient of Moment Surrogate Plot')    
+        plt_handle = plt.contourf(AoA_mesh/Units.deg,mach_mesh,CM_sur,levels=None)
+        cbar = plt.colorbar()
```
Comment (WalterMaier) [inline trunk/SUAVE/Analyses/Stability/AVL.py]: Same here

File: trunk/SUAVE/Analyses/Stability/AVL.py [hunk 5]
```diff
@@ -442,11 +442,21 @@ def build_surrogate(self):
 
         for jj in range(len(AoA_points)):
             for ii in range(len(mach_points)):
-                CM_sur[ii,jj]    = cm_surrogate.predict(np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]]))
-                Cm_a_sur[ii,jj]  = cm_alpha_surrogate.predict(np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]]))
-                Cn_b_sur[ii,jj]  = cn_beta_surrogate.predict(np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]]))
-                NP_sur[ii,jj]    = neutral_point_surrogate.predict(np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]]))                
+                CM_sur[ii,jj]    = cm_surrogate.predict([np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]])])
+                Cm_a_sur[ii,jj]  = cm_alpha_surrogate.predict([np.array([AoA_mesh[ii,jj],mach_mesh[ii,jj]])])
```
Comment (WalterMaier) [inline trunk/SUAVE/Analyses/Stability/AVL.py]: What is this?

File: regression/scripts/AVL/avl_files/aircraft.avl (hunk 1 above)
Comment (jmvegh) [reply regression/scripts/AVL/avl_files/aircraft.avl]: Yeah, I think something weird might have happened merging the latest version of develop. I'll use the old values unless @mclarke2  wants to stick with the newer values

File: regression/scripts/sizing_loop/sizing_loop.py (hunk 2 above)
Comment (jmvegh) [reply regression/scripts/sizing_loop/sizing_loop.py]: Regression failed on my laptop using the 1e-6 tolerance; may be an OS issue

File: trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py (hunk 3 above)
Comment (jmvegh) [reply trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py]: May have been a Git Merge issue. I'll take out

File: trunk/SUAVE/Analyses/Stability/AVL.py (hunk 4 above)
Comment (jmvegh) [reply trunk/SUAVE/Analyses/Stability/AVL.py]: May have been a Git Merge issue. I'll take out

File: trunk/SUAVE/Analyses/Stability/AVL.py (hunk 5 above)
Comment (jmvegh) [reply trunk/SUAVE/Analyses/Stability/AVL.py]: May have been a Git Merge issue. I'll take out

File: trunk/SUAVE/Sizing/write_sizing_residuals.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+## @ingroup Sizing
+#write_sizing_residuals.py
+# Created : May 2018, M. Vegh
+
+
```
Comment (WalterMaier) [inline trunk/SUAVE/Sizing/write_sizing_residuals.py]: extra space

File: trunk/SUAVE/Sizing/write_sizing_residuals.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+# ----------------------------------------------------------------------
+#  write_sizing_residuals
+# ----------------------------------------------------------------------
+
+
```
Comment (WalterMaier) [inline trunk/SUAVE/Sizing/write_sizing_residuals.py]: extra space

File: regression/scripts/AVL/avl_files/aircraft.avl (hunk 1 above)
Comment (mclarke2) [reply regression/scripts/AVL/avl_files/aircraft.avl]: It is a vertical stabilizer, the y value should be 0 so I corrected it.

File: trunk/SUAVE/Sizing/Sizing_Loop.py [hunk 8]
```diff
@@ -37,29 +40,50 @@ def __defaults__(self):
...
+        self.write_residuals       = False  #set to True to write the residuals at every iteration
+        self.residual_filename     = 'y_err_values.txt'
         
         #parameters that may only apply to certain methods
         self.iteration_options     = Data()
         self.iteration_options.newton_raphson_tolerance          = 5E-2             #threshhold of convergence when you start using newton raphson
         self.iteration_options.max_newton_raphson_tolerance      = 2E-3             #threshhold at which newton raphson is no longer used (to prevent overshoot and extra iterations)
         self.iteration_options.h                                 = 1E-6             #finite difference step for Newton iteration
         self.iteration_options.initialize_jacobian               = 'newton-raphson' #how Jacobian is initialized for broyden; newton-raphson by default
+        self.iteration_options.jacobian                          = np.array([np.nan])
```
Comment (planes) [inline trunk/SUAVE/Sizing/Sizing_Loop.py]: why is this defaulted to nan?

File: trunk/SUAVE/Sizing/Sizing_Loop.py [hunk 9]
```diff
@@ -90,8 +114,8 @@ def evaluate(self, nexus):
         
         #initialize
         converged         = 0     #marker to tell if it's converged
-        j=0  #major iterations
-        i=0  #function evals
+        #j = 0  #major iterations
```
Comment (planes) [inline trunk/SUAVE/Sizing/Sizing_Loop.py]: delete

File: trunk/SUAVE/Sizing/Sizing_Loop.py (hunk 8 above)
Comment (jmvegh) [reply trunk/SUAVE/Sizing/Sizing_Loop.py]: I wanted to make sure that the code breaks if the Jacobian wasn't properly initialized

**Pull request merged**